### ISSUE TITLE
TypeError: coercing to Unicode: need string or buffer, NoneType found

### ISSUE THREAD

**Issue by Germione** [kennethreitz/pipenv#1043]
Hi,

I'm trying to install a local setup.py into my virtual environment/Pipfile:

I get this error : TypeError: coercing to Unicode: need string or buffer, NoneType found

Do you have any idea of the problem ?

Thank you in advance

##### Describe your environment

Mac OS X - Sierra
Python version: 3.6.3
Pipenv version: 8.3.1

##### Actual result
```bash
pipenv install -e .
```
```
Installing -e .…
Traceback (most recent call last):
  File "/usr/local/Cellar/pipenv/8.3.1/libexec/bin/pipenv", line 11, in <module>
    load_entry_point('pipenv==8.3.1', 'console_scripts', 'pipenv')()
  File "/usr/local/Cellar/pipenv/8.3.1/libexec/lib/python2.7/site-packages/pipenv/vendor/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/Cellar/pipenv/8.3.1/libexec/lib/python2.7/site-packages/pipenv/vendor/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/usr/local/Cellar/pipenv/8.3.1/libexec/lib/python2.7/site-packages/pipenv/vendor/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/Cellar/pipenv/8.3.1/libexec/lib/python2.7/site-packages/pipenv/vendor/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/Cellar/pipenv/8.3.1/libexec/lib/python2.7/site-packages/pipenv/vendor/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/Cellar/pipenv/8.3.1/libexec/lib/python2.7/site-packages/pipenv/cli.py", line 1869, in install
    converted = convert_deps_from_pip(package_name)
  File "/usr/local/Cellar/pipenv/8.3.1/libexec/lib/python2.7/site-packages/pipenv/utils.py", line 540, in convert_deps_from_pip
    os.path.isfile(req.name)) and not req.vcs:
  File "/usr/local/Cellar/pipenv/8.3.1/libexec/lib/python2.7/genericpath.py", line 37, in isfile
    st = os.stat(path)
TypeError: coercing to Unicode: need string or buffer, NoneType found
```
##### Steps to replicate

```bash
brew install pipenv
pipenv --python 3.6.3
pipenv install --dev
```
Pipfile

```
[[source]]

url = "https://pypi.python.org/simple"
verify_ssl = true
name = "pypi"

[dev-packages]

keyring = "*"
twine = "*"
"flake8" = "*"
pytest = "*"
"pytest-cov" = "*"

[packages]

requests = "*"
click = "*"
```

**Comment by techalchemy** [comment]
@Germione did you run `pipenv install --dev` or did you run `pipenv install -e .` in order to get this error message? 

If attempting the latter, try encapsulating it in quotes: `pipenv install '-e .'`

**Comment by craigloftus** [comment]
I'm getting the same error with `pipenv install -e ~/wagtail-extension`, bisecting the versions points to the problem being introduced between `8.2.7` and `8.3.0`.

Interestingly, encapsulating like `pipenv install "-e /home/craig/wagtail-extensions"` will cause the cli help to print, where as `pipenv install "-e ~/wagtail-extensions"` will trigger the reported error.

I also get the error if installing the above package on its own, like `pipenv install --three -e .` and `pipenv install --three '-e .'`.

**Comment by craigloftus** [comment]
The regression seems to have been introduced in https://github.com/kennethreitz/pipenv/commit/0a48fa090e2535041e5280511a6a2f1696de6797 which is part of https://github.com/kennethreitz/pipenv/pull/926.

**Comment by techalchemy** [comment]
#@craigloftus thanks for taking the time to dig into that! We actually have a fix implemented already (see #958) that corrects that regression, I’ll see about getting it merged soon

**Comment by techalchemy** [comment]
In the meantime for those of you trying to install the local project root as a package you can use the advice here: https://github.com/kennethreitz/pipenv/issues/1012#issuecomment-342898222

**Comment by bofm** [comment]
On Python 3.6 the error is:

`TypeError: stat: path should be string, bytes, os.PathLike or integer, not NoneType`

(added this comment for searchability)

**Comment by Germione** [comment]
@techalchemy @craigloftus : Thanks for your answers
And thanks for the advice: It's working fine !!

**Comment by saif-freestar** [comment]
In the meantime, I hacked it in an ugly way, and did `pip install -e .` to get over it.

**Comment by techalchemy** [comment]
@bofm thanks for the searchtext :p maybe will reduce the number of issues on this.  @Germione thank @AlJohri since it's his fault we even implemented local file install support (#540) and he is still dropping knowledge bombs to support it!

**Comment by techalchemy** [comment]
This is now resolved in #958
